wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -235,6 +238,29 @@
     private final GitUserResolver   userResolver;
     private final Repository        repository;
 
+    // setup authenticators to use JGit behind the proxy
+    private static Authenticator httpsProxyAuth;
+    private static Authenticator httpProxyAuth;
+    static {
+        if (nonNull(System.getProperty("http.proxyUser")) && nonNull(System.getProperty("http.proxyPassword"))) {
```

Comment:
1. I don't see any reason to use `nonNull(...)` against `== null`.
2. I don't think that camel cased properties are good choice, i would  preferable change them to `http.proxy_user` and `http.proxy_password`
3. `Authenticator` looks not good as anonymous class to me, i think you  should extract it to the private static _NAMED_ inner class, and it seems that findbugs  will suggest you to do the same.